examples/beacon_server.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright 2015 MaidSafe.net limited
+//
+// This MaidSafe Software is licensed to you under (1) the MaidSafe.net Commercial License,
+// version 1.0 or later, or (2) The General Public License (GPL), version 3, depending on which
+// licence you accepted on initial access to the Software (the "Licences").
+//
+// By contributing code to the MaidSafe Software, or to this project generally, you agree to be
+// bound by the terms of the MaidSafe Contributor Agreement, version 1.0, found in the root
+// directory of this project at LICENSE, COPYING and CONTRIBUTOR respectively and also
+// available at: http://www.maidsafe.net/licenses
+//
+// Unless required by applicable law or agreed to in writing, the MaidSafe Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, either express or implied.
+//
+// See the Licences for the specific language governing permissions and limitations relating to
+// use of the MaidSafe Software.
+
+extern crate crust;
+
+use std::thread::spawn;
+use std::net::UdpSocket;
+
+fn main() {
+  spawn(move || {
+    let normal_socket = match UdpSocket::bind("::0:0") {
+      Ok(socket) => socket,
+      Err(e) => panic!("Couldn't bind socket: {}", e)
+    };
+
+    match normal_socket.local_addr() {
+      Ok(local_addr) => crust::beacon::listen_for_broadcast(local_addr),
```

Comment:
I don't seem to reach this point, neither do I get panics;  are panics in a spawned off thread invisible on terminal?